crates/router/src/services/api.rs

```diff
@@ -381,15 +381,25 @@ where
                     let request_method = request.method;
 
                     let current_time = Instant::now();
-                    let response = call_connector_api(state, request).await;
+                    let mut response = call_connector_api(state, request).await;
```

Comment:
why is `mut` needed here?

Reply:
have updated it by changing as_mute to as_ref